lib/containers/utils.pm

```diff
@@ -154,23 +154,31 @@ sub basic_container_tests {
 # Setup environment
 sub container_set_up {
     my ($dir, $file, $base) = @_;
-    die "You must define the directory!"  unless $dir;
-    die "You must define the Dockerfile!" unless $file;
-    my $basename_expected = script_run("grep baseimage_var $dir/BuildTest/$file");
-    die "Base image name is required for $file" if !$basename_expected && $base;
+    if (!defined($dir) || !!($dir =~ /^ *$/)) {
+        die 'You must define build directory in SUT!';
+    }
+    elsif (!defined($file) || !!($file =~ /^ *$/)) {
+        die 'You must define a template Dockerfile from data/containers/ directory!';
+    }
+    else {
+        assert_script_run "mkdir -p $dir/BuildTest/templates";
+    }
```

Comment:
is this section an improvement or you also saw something failing here?

Reply:
These 2 lines does not work ( or at least I haven't found and use case ) 
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/containers/utils.pm#L160

The file https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/containers/utils.pm#L159 should be just pulled later over here https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/containers/utils.pm#L166. 

And the other point, the tests were testing only defined-ness, so if there is any caller code that dynamically determines arguments for `container_set_up` and it will pass an empty string, it would not die.

Reply:
Returned, sorry for noise :zany_face: